File: MainRiceCooker_test.go
package main

import (
	"testing"
)

func TestRiceCooker(t *testing.T) {
	cooker := &RiceCooker{}

	// Test setupCooking
	cooker.setupCooking(2, 4, 100, 10)
	if cooker.quantityOfRice != 2 || cooker.quantityOfWater != 4 || cooker.temperature != 100 || cooker.timer != 10 {
		t.Errorf("setupCooking failed. Expected values not set.")
	}

	// Test startCooking
	cooker.startCooking()
	if !cooker.isCooking || cooker.currentStatus != "Cooking started. Timer set for 10 minutes." {
		t.Errorf("startCooking failed. Cooker not started.")
	}

	// Test warmMode
	cooker.warmMode()
	if cooker.currentTemperature != 60 || cooker.currentStatus != "Warm mode activated. Temperature set to 60°C." {
		t.Errorf("warmMode failed. Warm mode not activated.")
	}

	// Test pauseCooking
	cooker.pauseCooking()
	if cooker.isCooking || cooker.currentStatus != "Cooking paused." {
		t.Errorf("pauseCooking failed. Cooker not paused.")
	}

	// Test resumeCooking
	cooker.resumeCooking()
	if !cooker.isCooking || cooker.currentTemperature != 100 || cooker.currentStatus != "Cooking resumed. Timer set for 10 minutes." {
		t.Errorf("resumeCooking failed. Cooker not resumed.")
	}

	// Test checkStatus
	cooker.checkStatus() // Just check if it runs without errors, as output can vary

	// Test logError
	logError("test error message") // Just check if it runs without errors
}

File: src/MainRiceCooker.test.ts
import { RiceCooker } from "./MainRiceCooker";
describe('RiceCooker', () => {
  let cooker: RiceCooker;

  beforeEach(() => {
    cooker = new RiceCooker();
  });

  test('should set up cooking', () => {
    cooker.setupCooking(1, 2, 80, 1);

    expect(cooker['quantityOfRice']).toBe(1);
    expect(cooker['quantityOfWater']).toBe(2);
    expect(cooker['currentTemperature']).toBe(80);
    expect(cooker['timer']).toBe(1);
    expect(cooker['currentStatus']).toBe('Setup completed. Ready to start cooking.');
  });

  test('should start cooking', () => {
    cooker.setupCooking(1, 2, 80, 1);
    cooker.startCooking();

    expect(cooker['isCooking']).toBe(true);
    expect(typeof cooker['currentTimer']).toBe('object');
    expect(cooker['currentStatus']).toContain('Cooking started. Timer set for');
  });

  test('should pause cooking', () => {
    cooker.setupCooking(1, 2, 80, 1);
    cooker.startCooking();
    cooker.pauseCooking();

    expect(cooker['isCooking']).toBe(false);
    expect(cooker['currentTemperature']).toBe(0);
    expect(cooker['currentStatus']).toBe('Cooking paused.');
  });

  test('should resume cooking', () => {
    cooker.setupCooking(1, 2, 80, 1);
    cooker.startCooking();
    const originalTimer = cooker['currentTimer'];
    cooker.pauseCooking();
    cooker.resumeCooking();

    expect(cooker['isCooking']).toBe(true);
    expect(typeof cooker['currentTimer']).toBe('object');
    expect(cooker['currentTimer']).not.toBe(originalTimer);
    expect(cooker['currentStatus']).toContain('Cooking resumed. Timer set for');
  });

  test('should activate warm mode', () => {
    cooker.setupCooking(1, 2, 80, 1);
    cooker.warmMode();

    expect(cooker['currentTemperature']).toBe(60);
    expect(cooker['currentStatus']).toBe('Warm mode activated. Temperature set to 60°C.');
  });

  test('should check status', () => {
    cooker.setupCooking(1, 2, 80, 1);
    const consoleSpy = jest.spyOn(console, 'log');
    cooker.checkStatus();

    expect(consoleSpy).toHaveBeenCalledWith(expect.stringContaining('Status:'));
    expect(consoleSpy).toHaveBeenCalledWith(expect.stringContaining('Temperature:'));
  });
});
